wsmatrix@martin.zurowietz.de/overview/thumbnailsBox.js

```diff
@@ -0,0 +1,351 @@
+const WorkspaceThumbnail = Self.imports.workspacePopup.workspaceThumbnail;
+const Util = Self.imports.util;
+
+// The maximum size of a thumbnail is 5% the width and height of the screen
+var MAX_THUMBNAIL_SCALE = 0.05;
+var ThumbnailState = {
+    NEW: 0,
+    EXPANDING: 1,
+    EXPANDED: 2,
+    ANIMATING_IN: 3,
+    NORMAL: 4,
+    REMOVING: 5,
+    ANIMATING_OUT: 6,
+    ANIMATED_OUT: 7,
+    COLLAPSING: 8,
+    DESTROYED: 9,
+};
+
+
+var ThumbnailsBox = class {
+    constructor(settings, keybindings) {
+        this._settings = settings;
+        this._keybindings = keybindings;
+
+        this._overrideProperties = {
+            addThumbnails(start, count) {
+                let workspaceManager = global.workspace_manager;
+
+                for (let k = start; k < start + count; k++) {
+                    let metaWorkspace = workspaceManager.get_workspace_by_index(k);
+                    let thumbnail = new WorkspaceThumbnail.WorkspaceThumbnail(metaWorkspace, this._monitorIndex);
+                    thumbnail.setPorthole(this._porthole.x, this._porthole.y,
+                        this._porthole.width, this._porthole.height);
+                    this._thumbnails.push(thumbnail);
+                    this.add_actor(thumbnail);
+
+                    if (this._shouldShow && start > 0 && this._spliceIndex === -1) {
+                        // not the initial fill, and not splicing via DND
+                        thumbnail.state = ThumbnailState.NEW;
+                        thumbnail.slide_position = 1; // start slid out
+                        thumbnail.collapse_fraction = 1; // start fully collapsed
+                        this._haveNewThumbnails = true;
+                    } else {
+                        thumbnail.state = ThumbnailState.NORMAL;
+                    }
+
+                    this._stateCounts[thumbnail.state]++;
+                }
+
+                this._queueUpdateStates();
+
+                // The thumbnails indicator actually needs to be on top of the thumbnails
+                this.set_child_above_sibling(this._indicator, null);
+
+                // Clear the splice index, we got the message
+                this._spliceIndex = -1;
+            },
+
+            vfunc_get_preferred_height(forWidth) {
+                let workspaceManager = global.workspace_manager;
+                let rows = workspaceManager.layout_rows;
+                let columns = workspaceManager.layout_columns;
+                let themeNode = this.get_theme_node();
+
+                forWidth = themeNode.adjust_for_width(forWidth);
+
+                let spacing = themeNode.get_length('spacing');
+                let totalSpacing = (rows - 1) * spacing;
+
+                const avail = forWidth - totalSpacing;
+
+                let scale = (avail / columns) / this._porthole.width;
+                scale = Math.min(scale, MAX_THUMBNAIL_SCALE);
+
+                const height = Math.round(this._porthole.height * scale) * 5;
```

Comment:
Where is the 5 coming from and what's it for?

Reply:
I am actually not sure about this one

Reply:
This doesn't seem to have any effect. I tried `5`, `50`, `500` and `0` and only with `0` the thumbnails box in the overview disappeared. The other values all produced the same look.